03-Modify-Raster.Rmd

```diff
@@ -74,197 +72,240 @@ Read more about the `raster` package in R.</a>
 
-`functionName <- function(variable1, variable2){
+##Writing Functions in R
+Before we can continue with `overlay()` we need to discuss writing functions. A 
+function represents a set of tasks that need to be repeated over and over. 
 
+LINK TO MORE ABOUT FUNCTIONS IN R 
+I like explanations (how to AND why) here: http://nicercode.github.io/guides/functions/
+but I'm hesitant to link to blogs that might disappear so also QuickR link:
+http://www.statmethods.net/management/userfunctions.html
+Thoughts? 
 
-}
+A simplified syntax for a function in R is as follows:
 
-LINK TO MORE ABOUT FUNCTIONS IN R 
+`functionName <- function(variable1, variable2){WhatYouWantDone, WhatToReturn}`
 
+Let's say we want to create a function that subtracts one raster from another. 
+In our case what we want does is what we want returned so we don't have to 
+specify it twice. 
 ```{r create-function }
-
-#first, let's create a function
 #this function will take two rasters (r1 and r2) and subtract them
 subtRasters <- function(r1, r2){
-return(r1-r2)
+(r1-r2)
 }
-
 ```
 
+We now have a function called `subtRasters` that will subtract any two rasters
+that we give it.  
 
+##Continuing with Overlay Function
+Let's repeat our simple raster math computation using `overlay()` and the
+function we just created. 
 ```{r raster-overlay }
+CHM_ov<- overlay(DSM,DTM,fun=subtRasters)
 
-
-CHM_ov_HARV <- overlay(DSM,DTM,fun=subtRasters)
-
-plot(CHM_ov_HARV,
+plot(CHM_ov,
      main="NEON Canopy Height Model - Overlay Subtract\n Harvard Forest")
-
 ```
 
-So now, we've learned two different ways to subtract two rasters in R.
-Once we have an output raster, we can use the `writeRaster` function to export
-the object as a new geotiff. 
+#Writing Raster Files
+So now we've learned two different ways to do calculations (in our case 
+subtraction) with two rasters in R.  This gives us an output raster object 
+(`CHM_ov`), we can use the `writeRaster` function to export the object as a new 
+geotiff. 
+
+>NAMES: Part of writing a raster (or any file) is choosing a suitable name for 
+>the file.  File names should be make it clear what is in the file.  This is 
+>also true with object names in R.  In fact, so far in these lessons we've been 
+>using poor nameing practices as we haven't been including any information about
+>the location in our names (eg. `DSM` not `DSM_HARV`).  This hasn't gotten us 
+>in trouble as we've only been working in one location (Harvard Forest), 
+>however, it is still good practice to be specific with file and object names.
+
+When we write this raster object to a GeoTIFF file we'll name it 
+`chm_ov_HARV.tiff` this way we know that it is a canopy height model of Harvard 
+Forest created using overlay.  Any files you write will be saved in whichever 
+file is the current working directory. 
 
 ```{r write-raster }
-
-#export CHM object to new geotiff
-writeRaster(CHM_ov_HARV,"chm_ov_HARV.tiff",
+#export CHM object to new GeotIFF
```

Comment:
I like this section in the write raster - it's good to talk about naming practices!